Documentation/Examples/RetriableSample.dib

```diff
@@ -0,0 +1,660 @@
+#!markdown
+
+# Corvus.Retry.Retriable - Sample Notebook
+
+#!markdown
+
+First, add a reference to the `Corvus.Retry` NuGet package.
+
+#!csharp
+
+#r "nuget: Corvus.Retry, 1.0.2"
+
+#!markdown
+
+Add a using statement for `Corvus.Retry` namespace and the others we'll be using in this notebook.
+
+#!csharp
+
+using Corvus.Retry;
+using Corvus.Retry.Strategies;
+using Corvus.Retry.Policies;
+using System.Net.NetworkInformation;
+using System.Threading;
+using System.Net.Http;
+using System.Web;
+using System.Diagnostics;
+using System.Threading;
+
+#!markdown
+
+`Retriable` is the most common usage pattern from the `Corvus.Retry` library. It wraps an existing method and allows you to re-execute the method if it throws an exception.
+
+```csharp
+Retriable.Retry(() => DoSomethingThatMightFail());
+```
+
+#!markdown
+
+### `Retriable.Retry` – basic example
+
+The following example uses `Retry` to wrap a method that pings an internet host.
+
+#!csharp
+
+Ping pingSender = new Ping();
+Retriable.Retry(() => pingSender.Send("www.endjin.com"));
+
+#!markdown
+
+You can also return the values from the method. The following example uses the `PingReply` object returned by `Ping.Send()`:
+
+#!csharp
+
+Ping pingSender = new Ping();
+PingReply reply = Retriable.Retry(() => pingSender.Send("www.endjin.com"));
+
+// Print a property from the response
+Console.WriteLine($"Address:\t{reply.Address.ToString()}");
+
+#!markdown
+
+### `Retriable.RetryAsync` – basic example
+
+The `RetryAsync` method allows you to wrap an asynchronous method to re-execute it.
+
+#!markdown
+
+The following example uses `Retriable` to wrap an asynchronous method that sends a ping to a internet host and receives a response.
+
+#!csharp
+
+Ping pingSender = new Ping();
+PingReply reply = await Retriable.RetryAsync(() => pingSender.SendPingAsync("www.endjin.com"));
+
+// Print a property from the response
+Console.WriteLine($"Address:\t{reply.Address.ToString()}");
+
+#!markdown
+
+## Cancellation token
+
+You can pass a cancellation token to `Retry` to signal cancellation of the retry process.
+
+### Using a cancellation token - example
+
+The default behavior of `Retry` is to not use a cancellation token: `CancellationToken.None`.
+
+#!csharp
+
+Ping pingSender = new Ping();
```

Comment:
This example really needs a `CancellationTokenSource` and an example of you cancelling before it is finished retrying (with the Task as per the example we talked about the other day)

Reply:
I've added a new example that shows this.